portfolio/src/main/webapp/script.js

```diff
@@ -81,12 +81,48 @@ function createCommentElement(comment) {
   commentElement.classname = 'comment';
 
   const headerElement = document.createElement('span');
-  headerElement.innerHTML = "<h3>"+comment.posterName +"<h3>";
+  headerElement.innerHTML = "<h3>" + cleanseString(comment.posterName) +"<h3>";
 
   const bodyElement = document.createElement('span');
-  bodyElement.innerHTML = "<p>" + comment.comment +"<p>";
+  bodyElement.innerHTML = "<p>" + cleanseString(comment.comment) +"<p>";
 
   commentElement.appendChild(headerElement);
   commentElement.appendChild(bodyElement);
   return commentElement;
-}
\ No newline at end of file
+}
+
+function cleanseString(html) {
+  // Prevent html injection by replacing escape characters with plaintext
+  return html.replace(/</g, "&lt;").replace(/>/g, "&gt;");
+}
+
+google.charts.load('current', {'packages': ['corechart']});
+google.charts.setOnLoadCallback(drawChart);
+
+function drawChart() {
+  const data = new google.visualization.DataTable();
+  data.addColumn('string', 'Poster');
+  data.addColumn('number', 'Comment Length');
+  var dropDown = document.getElementById('numCommentsDropDown');
+  var commentLimit = dropDown.options[dropDown.selectedIndex].value;
+  // Grabs comments to consider based off dropdown value.
+  fetch('/data?num-comments='+commentLimit).then(response => response.json()).then((comments) =>{
```

Comment:
given that you use this fetch multiple times. Refactor it into a function. Use Async/Await to clean it up

Reply:
I'm not entirely sure how to do this without destabilizing the functionality.

Reply:
schedule some time with me and I can walk you through using Async/Await to clean this up.